### ISSUE TITLE
Incorrect `greatest` and `least` functions nullable behaviour

### ISSUE THREAD

**Issue by kongul** [prestosql/presto#5783]
We are updating Presto from version 326 to 345 and found that `greatest` and `least` functions nullable behaviour changed.

Un Presto 326
```
select least(null, 1, 2); -- returns null
select greatest(null, 1, 2); -- returns null
```

Un Presto 345
```
select least(null, 1, 2); -- returns 1
select greatest(null, 1, 2); -- returns 2
```

As it said in the doc https://prestosql.io/docs/current/functions/comparison.html#greatest-and-least , they should return null if one of the arguments is null
We rely on this behaviour in our sql scripts

**Comment by martint** [comment]
Thanks for the report. That is a regression, indeed. I'm investigating.

**Comment by martint** [comment]
The bug was introduced in 344. I'm tracking down the exact commit now.

**Comment by martint** [comment]
This is the commit that introduced the issue: c95e937517f773ba625cfde38aac7ed11e6f8493

cc @dain

**Comment by electrum** [comment]
That commit also broke the behavior for `array_min` and `array_max` for NULL and NaN.

**Comment by electrum** [comment]
NaN behavior is wrong for `min` and `max`:
```
presto> select min(x), max(x) from (values 1, 3, 5, nan()) as t (x);
 _col0 | _col1 
-------+-------
   1.0 |   NaN 
(1 row)
```

**Comment by findepi** [comment]
> That commit also broke the behavior for array_min and array_max for NULL and NaN.

> NaN behavior is wrong for min and max:

@electrum  Let's make sure this is covered by an issue (either new one, or expand description of this one).

**Comment by findepi** [comment]
> NaN behavior is wrong for `min` and `max`:
> 
> ```
> presto> select min(x), max(x) from (values 1, 3, 5, nan()) as t (x);
>  _col0 | _col1 
> -------+-------
>    1.0 |   NaN 
> ```

This behavior is also change (between 343 and 344). Results for 343:
```
presto> select min(x), max(x) from (values 1, 3, 5, nan()) as t (x);
 _col0 | _col1
-------+-------
   1.0 |   5.0
```

**Comment by electrum** [comment]
https://github.com/prestosql/presto/issues/5828

**Comment by dain** [comment]
I believe the null behavior of greatest and least was fixed iin #5787

**Comment by findepi** [comment]
@dain thanks for a quick turn around. 
Do we have issues for other things mentioned by @electrum in https://github.com/prestosql/presto/issues/5783#issuecomment-720735877 and https://github.com/prestosql/presto/issues/5783#issuecomment-720788322?
